### PR TITLE
Validate with libtidy on travis

### PR DESCRIPTION
**Pull request by spk**
Use libtidy https://github.com/htacg/tidy-html5
for complete validation of the website, just excluding `examples` directory (inserted in javascript).
Also update validate-website gem.

Time ~ 40sec for `check:markup` cf https://travis-ci.org/ruby/www.ruby-lang.org/builds/501125031#L582

Cheers

### CODE REVIEW

**Comment by stomar** [comment]
HTML validation on Travis has been discussed before, I think.

IMO it's a bad idea to check every build:

* it's really not necessary since the HTML layouts very rarely change; most commits only add/change markdown, which usually does not introduce malformed HTML
* it takes way too much time (the build of this PR took **8 min 22 s**); note that there already have been complaints about the current 2...3 min

I rather think that occasional validations should be sufficient.

**Comment by spk** [comment]
Hi, I've updated the PR using libtidy (around 40sec to check all files) and fixes the errors hope we could reconsider including the changes, thanks

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,14 @@
+---
```
Comment (stomar) [inline .travis.yml]: unrelated change

```suggestion
```

**Comment by stomar** [comment]
It‘s a little confusing: this is supposed to target CI, but commit message says „locally“.

File: .travis.yml (hunk 1 above)
Comment (spk) [reply .travis.yml]: Its yaml headers recommended by linters should i remove it ?

**Comment by spk** [comment]
> It‘s a little confusing: this is supposed to target CI, but commit message says „locally“.

you are right its confusing i will remove this

File: .travis.yml (hunk 1 above)
Comment (spk) [reply .travis.yml]: Removed in https://github.com/ruby/www.ruby-lang.org/pull/1420/commits/fea533eb88543172cf6acc8842b2efebd7f96db0

**Comment by stomar** [comment]
After a closer look I feel I have to close this PR, since it doesn‘t make any sense to run on CI in this form, and would even be harmful.

The problem is that the validation task does not check the current build but the already deployed site. This would mean that a PR introducing errors would **not** fail on CI, but all subsequent PRs **would**, regardless of whether they introduce errors or not.

**Comment by stomar** [comment]
As to the general question whether this check should (in a fixed version) be added to CI: 40 seconds is not too bad, but personally I’m still undecided and would like to have some other opinions.

@hsbt

**Comment by spk** [comment]
> The problem is that the validation task does not check the current build but the already deployed site. This would mean that a PR introducing errors would not fail on CI, but all subsequent PRs would, regardless of whether they introduce errors or not.

This is **not** correct this test the current branch since it check the build-ed `_site` generated on CI. cf https://github.com/ruby/www.ruby-lang.org/blob/master/lib/markup_checker.rb#L7

**Comment by spk** [comment]
> As to the general question whether this check should (in a fixed version) be added to CI: 40 seconds is not too bad, but personally I’m still undecided and would like to have some other opinions.
> 
> @hsbt

If this something the team should talk about could you please reopen the PR @stomar please ?

**Comment by spk** [comment]
Reopened in https://github.com/ruby/www.ruby-lang.org/pull/1990 to have team feedback

**Comment by stomar** [comment]
Then please remind me what is the meaning of the `—site https://www.ruby-lang.org/` option?
